src/contact.rs

```diff
@@ -983,6 +986,24 @@ pub fn addr_normalize(addr: &str) -> &str {
     norm
 }
 
+fn sanitize_name_and_addr(name: impl AsRef<str>, addr: impl AsRef<str>) -> (String, String) {
+    lazy_static! {
+        static ref ADDR_WITH_NAME_REGEX: Regex = Regex::new("(.*)<(.*)>").unwrap();
+    }
+    if let Some(captures) = ADDR_WITH_NAME_REGEX.captures(addr.as_ref()) {
```

Comment:
The shortest way I found to do this was: 
```
    if let Some(captures) = ADDR_WITH_NAME_REGEX.captures(addr.as_ref()) {
        if let [n, a] = &captures.iter().collect::<Vec<_>>()[..] {
...
```
because 
- it is not that easy to build a slice from a `Captures` struct
- a slice is not owned, so I can't use `map` but need the two `if`s

Reply:
So, I do not think this is worth it because it introduces quite some complexity (unless there is a way to express this shorter I did not find)